### ISSUE TITLE
Back button broken in Chrome when setting location.href in HTML5 mode

### ISSUE THREAD

**Issue by circlingthesun** [angular/angular.js#11667]
When clicking a link from ```http://example.com/``` to ```http://example.com/app``` where ```/app``` runs angular with HTML5 mode and the baseURL set to ```/app```, the following error is raised when you press the back button:

```Error: [$location:ipthprfx] Invalid url "http://example.com/", missing path prefix "http://example.com/app/"```

The offending line:
```var pathUrl = beginsWith(appBaseNoFile, url);```

Can be found here:
https://github.com/angular/angular.js/blob/a509e9aa149d0f88cc39f703d539f7ffd4cd6103/src/ng/location.js#L107

```pathUrl``` returns undefined which results in an error being thrown and the page not navigating back to ```http://example.com/``` as expected.

**Comment by bruunofco** [comment]
Hello,
Displays the script that you are using to return the page.
And send an example in Plunker your code so we can better help you.

**Comment by circlingthesun** [comment]
I've reproduced this here: http://public.jacklab.co.za/

**Comment by circlingthesun** [comment]
Click the link, then click back.

**Comment by circlingthesun** [comment]
In reproducing this I noticed that this only happens when the next page is navigated to by setting window.location.href. The reason for doing this is that the router at / sometimes intercepts links pointing to pages under the router at /app. Setting window.location.href allows me to bypass the router at /.

**Comment by circlingthesun** [comment]
Also, this error does not happen in Firefox. I've only confirmed the bug in chrome 41.

**Comment by Narretz** [comment]
@petebacondarwin I assigned you since you said you where looking into location issues, hope that's okay.

**Comment by petebacondarwin** [comment]
@circlingthesun - I think we may have fixed this bug in https://github.com/angular/angular.js/commit/6903b5ec4c04ed6b7c80ef7d638c48639ccdc4bb.

I copied down your reproduction of the issue to localhost but I couldn't get it to fail. Are you doing something clever on your server?

Can you try out the version at https://code.angularjs.org/snapshot?

**Comment by circlingthesun** [comment]
I just updated the code to point the snapshot but the error still happens. I'm not doing anything fancy on the server. The site resolves to index.html and /app resolves to app.html. My nginx config is as follows:

```
server {
    root /home/rickert/Public;
    server_name public.*;

    location / {
        autoindex on;
    }

    location /app {
        try_files /app.html =404;
    }
}
```

**Comment by petebacondarwin** [comment]
OK, so here is a failing Plunker: http://plnkr.co/edit/wQ7WxkIPTL1L8p5VnkxR?p=preview

**Comment by petebacondarwin** [comment]
If you use target="_self" then you don't need the `window.location.href` hack and there is no error.
See http://plnkr.co/edit/szmCMV1j8HWSM9ODdmb3?p=preview

**Comment by circlingthesun** [comment]
That'll work when there is a link. In my actual code base I have a login modal that redirects to /app. Is there perhaps another way to get around this?

**Comment by petebacondarwin** [comment]
@circlingthesun - can you take a look at my proposed fix: #12361

**Comment by circlingthesun** [comment]
This does appear to fix the issue. (see http://public.jacklab.co.za/)

**Comment by petebacondarwin** [comment]
Closed by 9e492c3
